File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/endpoints/datasync/relationship/UpdateRelationship.java
package com.pubnub.api.java.endpoints.datasync.relationship;

import com.pubnub.api.java.endpoints.Endpoint;
import com.pubnub.api.java.models.consumer.datasync.relationship.PNDataSyncUpdateRelationshipResult;
import org.jetbrains.annotations.Nullable;

/**
 * @see com.pubnub.api.java.datasync.DataSync#updateRelationship(String, java.util.List)
 */
public interface UpdateRelationship extends Endpoint<PNDataSyncUpdateRelationshipResult> {
    /**
     * Optional eTag for optimistic concurrency ({@code If-Match} header).
     */
    UpdateRelationship ifMatch(@Nullable String ifMatch);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/datasync/relationship/UpdateRelationship.kt
package com.pubnub.api.endpoints.datasync.relationship

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.datasync.relationship.PNDataSyncUpdateRelationshipResult

/**
 * @see [com.pubnub.api.datasync.DataSync.updateRelationship]
 */
actual interface UpdateRelationship : Endpoint<PNDataSyncUpdateRelationshipResult>
